### PR TITLE
Split spack functionality into setup and execute modules

### PR DESCRIPTION
**Pull request by nick-stroud**
This change:
- Moves functionality into spack-execute module
- Adds use of bucket & startup script to store spack scripts

Tested manually:
- Actual gromacs install from cache
- Adding data files 
- Chaining execute modules

Future changes:
- Rename spack install 
- Migrate examples
- Clean up spack install outputs
- update documentation

Notes on review:
- Ignore readmes (only automated changes)
- Note that most of code is just moving
- Review commits independently

### Submission Checklist

Please take the following actions before submitting this pull request.

* Fork your PR branch from the Toolkit "develop" branch (not main)
* Test all changes with pre-commit in a local branch [#](https://goo.gle/hpc-toolkit#development)
* Confirm that "make tests" passes all tests
* Add or modify unit tests to cover code changes
* Ensure that unit test coverage remains above 80%
* Update all applicable documentation
* Follow Cloud HPC Toolkit Contribution guidelines [#](https://goo.gle/hpc-toolkit-contributing)

### CODE REVIEW

File: community/modules/scripts/spack-execute/main.tf [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+      log_file   = var.log_file
+      commands   = local.commands_content
+    }
+  )
+
+  data_runners = [for data_file in var.data_files : merge(data_file, { type = "data" })]
+
+  execute_md5 = substr(md5(local.execute_contents), 0, 4)
+  execute_runner = {
+    "type"        = "ansible-local"
```
Comment (tpdownes) [inline community/modules/scripts/spack-execute/main.tf:38]: Remove the quotation marks from the key values (style convention we should uniformly adopt)

File: community/modules/scripts/spack-execute/versions.tf [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+terraform {
+  required_version = ">= 1.0.0"
+  required_providers {
+    local = {
+      source  = "hashicorp/local"
+      version = ">= 2.0.0"
```
Comment (tpdownes) [inline community/modules/scripts/spack-execute/versions.tf:22]: I'd feel better with a `~>` constraint here. The most recent breaking change was internal APIs of Terraform binary and plugins but we might anticipate something similar for Terraform 2.x

**Review by tpdownes — APPROVED** [review]
A couple minor change requests.

File: community/modules/scripts/spack-execute/main.tf (hunk 1 above)
Comment (nick-stroud) [reply community/modules/scripts/spack-execute/main.tf]: Done

File: community/modules/scripts/spack-execute/versions.tf (hunk 2 above)
Comment (nick-stroud) [reply community/modules/scripts/spack-execute/versions.tf]: Done.

**Pull request merged**